File: regexlib/2021-5-15/python_re_test_file/regexlib_7246.py
# 7246
# ^(\d)?[ ]*[\(\.\-]?(\d{3})[\)\.\-]?[ ]*(\d{3})[\.\- ]?(\d{4})[ ]*(x|ext\.?)?[ ]*(\d{1,7})?$
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:"1111111111"+" "*5000+"! _1_POA(i)"

import re
from time import perf_counter

regex = """^(\d)?[ ]*[\(\.\-]?(\d{3})[\)\.\-]?[ ]*(\d{3})[\.\- ]?(\d{4})[ ]*(x|ext\.?)?[ ]*(\d{1,7})?$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "1111111111" + " " * i * 10000 + "! _1_POA(i)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_7246.js
/* 7246
 * ^(\d)?[ ]*[\(\.\-]?(\d{3})[\)\.\-]?[ ]*(\d{3})[\.\- ]?(\d{4})[ ]*(x|ext\.?)?[ ]*(\d{1,7})?$
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:"1111111111"+" "*10000+"! _1_POA(i)"
 */
var REGEX = /^(\d)?[ ]*[\(\.\-]?(\d{3})[\)\.\-]?[ ]*(\d{3})[\.\- ]?(\d{4})[ ]*(x|ext\.?)?[ ]*(\d{1,7})?$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '1111111111' + ' '.repeat(i*10000) + '! _1_POA(i)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}